footbot/optimiser/team_selector.py

```diff
@@ -2,297 +2,391 @@
-    vice_selection_elements = list(player_elements[vice_selection_indices])
-
-    # get bench elements
-    bench_selection = [int(round(j)) for j in bench.value]
-    bench_selection_indices = [i for i, j in enumerate(bench_selection) if j == 1]
-    bench_selection_elements = list(player_elements[bench_selection_indices])
-
-    if existing_squad_elements:
-        transfers = {
-            "transfers_in": set(
-                first_team_selection_elements + bench_selection_elements
-            )
-            - set(existing_squad_elements),
-            "transfers_out": set(existing_squad_elements)
-            - set(first_team_selection_elements + bench_selection_elements),
-        }
-    else:
-        transfers = {"transfers_in": set(), "transfers_out": set()}
+    first_team = get_elements_from_vector(first_team_v.value, player_elements)
+    bench = get_elements_from_vector(bench_v.value, player_elements)
+    captain = get_elements_from_vector(captain_v.value, player_elements)
+    vice = get_elements_from_vector(vice_v.value, player_elements)
+    transfers = {
+        "transfers_in": set(first_team + bench) - set(existing_squad),
+        "transfers_out": set(existing_squad) - set(first_team + bench),
+    }
 
     return (
-        first_team_selection_elements,
-        captain_selection_elements,
-        vice_selection_elements,
-        bench_selection_elements,
+        first_team,
+        bench,
+        captain,
+        vice,
         transfers,
     )
 
 
-def optimise_entry(
-    entry,
-    total_budget=1000,
-    bench_factor=0.1,
-    transfer_penalty=4,
-    transfer_limit=15,
-    start_event=1,
-    end_event=38,
-    login=None,
-    password=None,
-):
+def get_private_entry_data(entry, login, password):
     """
-    optimise a given entry based on their picks
-    for the current event
+    Get private entry data using specified credentials.
+
+    Private entry data includes recent squad changes, players prices and bank balance.
+
+    :param entry: Team identifier
+    :param login: FPL login
+    :param password: FPL password
+    :return: Dictionary of private entry data
     """
 
-    logger.info("getting current event")
-    bootstrap_request = requests.get(
-        "https://fantasy.premierleague.com/api/bootstrap-static/"
-    )
-    bootstrap_data = bootstrap_request.json()
+    session = requests.session()
+
+    payload = {
+        "redirect_uri": "https://fantasy.premierleague.com/a/login",
+        "app": "plfpl-web",
+        "login": login,
+        "password": password,
+    }
 
-    # if no events are current, current event is zero
-    # season has yet to start
-    current_event = 0
-    for event in [i for i in bootstrap_data["events"] if i["is_current"]]:
-        # otherwise, take event id of event that is current
-        current_event = event["id"]
+    logger.info("authenticating for entry")
+    session.post("https://users.premierleague.com/accounts/login/", data=payload)
```

Comment:
```suggestion
    resp = session.post("https://users.premierleague.com/accounts/login/", data=payload)
    resp.raise_for_status()
```
now we get the real error (some `HTTPError`) if user/pass was wrong rather than returning junk